pkg/storage/db_test.go

```diff
@@ -0,0 +1,232 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/asecurityteam/asset-inventory-api/pkg/domain"
+	"github.com/asecurityteam/logevent"
+	"github.com/asecurityteam/runhttp"
+)
+
+func TestGracefulHandlingOfTxBeginFailure(t *testing.T) {
+	// no panics, in other words
+	mockdb, mock, err := sqlmock.New()
+	if err != nil {
+		t.Fatalf("an error '%s' was not expected when opening a stub database connection", err)
+	}
+	defer mockdb.Close()
+
+	thedb := DB{mockdb, sync.Once{}}
+
+	mock.ExpectBegin().WillReturnError(fmt.Errorf("could not start transaction"))
+
+	ctx := context.Background()
+	ctx = logevent.NewContext(ctx, stdoutLogger(ctx))
+
+	if err = thedb.StoreCloudAsset(ctx, fakeCloudAssetChanges()); err == nil {
+		t.Errorf("was expecting an error, but there was none")
+	}
+	assert.Equal(t, "could not start transaction", err.Error())
+
+	if err := mock.ExpectationsWereMet(); err != nil {
```

Comment:
`testify` has `assert.NotError()` and `require.NotError` as helpers for these cases.